sdk/keyvault/Azure.Security.KeyVault.Keys/src/KeyClient.cs

```diff
@@ -95,7 +95,9 @@ public virtual async Task<Response<Key>> CreateRsaKeyAsync(RsaKeyCreateOptions r
 
         public virtual Response<Key> UpdateKey(KeyBase key, IEnumerable<KeyOperations> keyOperations, CancellationToken cancellationToken = default)
         {
-            if (string.IsNullOrEmpty(key?.Version)) throw new ArgumentException($"{nameof(key.Version)} can't be empty or null");
+            if (key == null) throw new ArgumentNullException(nameof(key));
+            if (key.Version == null) throw new ArgumentNullException($"{nameof(key)}.{nameof(key.Version)}");
```

Comment:
Are empty strings allowed for key.Version?

Reply:
After the API review, they decided to leave that check to the server. In our side just make sure that there is an object for version